doc/develop/west/zephyr-cmds.rst

```diff
@@ -174,3 +174,20 @@ As does deleting them::
 Additionally the tool allows you to specify the modules you want to list,
 fetch or clean blobs for by typing the module names as a command-line
 parameter.
+
+.. _west-twister:
+
+Twister wrapper: ``west twister``
+*****************************************
+This command is a wrapper for :ref:`twister <twister_script>`.  It leverages
+current :envvar:`ZEPHYR_BASE` environment variable and west manifest to launch
+the twister version associated with the west workspace. `west twister` resolves
+path to twister in the following order
+
+#. :envvar:`ZEPHYR_BASE` environment variable
+#. west manifest config for twister.path (see :ref:`west-config-cmd` for more details)
+#. Current path of zephyr_ext_common.py
+
+To set the path of twister for current west workspace use::
+
+  west config twister.path <path_to_twister>
```

Comment:
Actually, thinking about this a bit more, is it really worth it to have a configuration option for the location of twister? If we don't have a use case for it already I would rather not add it.

Reply:
I initially tried using config option `zephyr.base` for this but when testing scenarios where this didn't exist yet, lead it to be automatically populated with 
```
[zephyr]
base = zephyr
```
So on my second call when testing `west twister` used the wrong version of twister. I couldn't find the code that was doing this, so I went with the more direct and unused `twister.path` option name.  
My preference to give this as an option to the developer is so that they don't have to keep updating `ZEPHYR_BASE` environment variable, when switching between projects.  If this is configured in their west workspace, just changing to the directory will update them to the right path for twister.  I'm open to other ideas to achieve this behavior.

Reply:
> My preference to give this as an option to the developer is so that they don't have to keep updating ZEPHYR_BASE environment variable, when switching between projects.

What ]application types](https://docs.zephyrproject.org/latest/develop/application/index.html#application-types) are you referring to in this scenario? I assume by "project" you mean "application". 

If you have multiple zephyr repository applications or zephyr workspace applications, there is no need to set ZEPHYR_BASE at all when building within the workspace, for example. (I'm assuming you build within the workspace directories in this case.)

If you have multiple freestanding applications that need to use different west workspaces, there is no avoiding resetting ZEPHYR_BASE when switching between them to build, is there? (I'm assuming you build within the application directory in this case.)

> If this is configured in their west workspace, just changing to the directory will update them to the right path for twister.

Assuming "the directory" corresponds to my above, I see no need for twister.path.

> I'm open to other ideas to achieve this behavior.

Seems like you can find twister relative to zephyr_ext_common's ZEPHYR_SCRIPTS without needing the config option.

Reply:
> This would be a [freestanding](https://docs.zephyrproject.org/latest/develop/application/index.html#zephyr-freestanding-application) application type. With a directory structure something like

What you have shown is not a freestanding application. A freestanding application is outside of any west workspace. You have shown app1 and app2 directories that are in fact the topdirs of independent west workspaces.